test/distributed/org/apache/cassandra/distributed/test/sai/SAIUtil.java

```diff
@@ -49,8 +51,20 @@ public static void waitForIndexQueryable(Cluster cluster, String keyspace)
     {
         assertGossipEnabled(cluster);
         final List<String> indexes = getIndexes(cluster, keyspace);
-        await().atMost(60, TimeUnit.SECONDS)
-               .untilAsserted(() -> assertIndexesQueryable(cluster, keyspace, indexes));
+        await().atMost(60, TimeUnit.SECONDS).pollInterval(1, TimeUnit.SECONDS)
+               .untilAsserted(() -> assertIndexStatus(cluster, keyspace, indexes, Index.Status.BUILD_SUCCEEDED));
+    }
+
+    /**
+     * Waits until all indexes in the given keyspace become non-queryable on a specific node.
+     */
+    public static void waitForIndexNonQueryable(Cluster cluster, String keyspace, int node)
+    {
+        assertGossipEnabled(cluster);
+        final List<String> indexes = getIndexes(cluster, keyspace);
+        InetAddressAndPort addressAndPort = nodeAddress(cluster.get(node).broadcastAddress());
+        await().atMost(60, TimeUnit.SECONDS).pollInterval(1, TimeUnit.SECONDS)
```

Comment:
The default polling interval seemed to be creating a lot of logging noise, so I bumped it to 1 second.